### PR TITLE
Remove deprecations: tock

### PR DESCRIPTION
**Pull request by caguero**
# 🦟 Bug fix

Perform the tock of the tick-tock cycle and remove / move deprecated code.

@peci1

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: Migration.md [hunk 1]
```diff
@@ -5,6 +5,38 @@ Deprecated code produces compile-time warnings. These warning serve as
...
+1. **gps_sensor.proto**
+    + This message is deprecated. Use `NavSat` instead.
+
+### Removals
+
+1. **spherical_coordinates.proto**
+    + The `LOCAL2` item has been removed from the 
+      `msgs::SphericalCoordinatesType` enum.
+
+1. **spherical_coordinates.hh**
```
Comment (peci1) [inline Migration.md:27]:
```suggestion
1. **SphericalCoordinates.hh**
```

**Review by peci1 — CHANGES_REQUESTED** [review]
Just one little suggestion, otherwise LGTM.

**Review by peci1 — APPROVED** [review]

File: Migration.md [hunk 2]
```diff
@@ -5,6 +5,41 @@ Deprecated code produces compile-time warnings. These warning serve as
...
+### Deprecations
+
+1. **camerasensor.proto**
+    + The `image_format` field is deprecated. Use `pixel_format` instead.
+
+1. **gps.proto**
+    + This message is deprecated. Use `NavSat` instead.
+
+1. **gps_sensor.proto**
+    + This message is deprecated. Use `NavSat` instead.
```
Comment (azeey) [inline Migration.md:19]:
```suggestion
    + This message is deprecated. Use `NavSatSensor` instead.
```

**Review by azeey — APPROVED** [review]
One small change, otherwise, LGTM!

**Pull request merged**